### PR TITLE
Fix: better HTML > MD conversion for editing, including lists and quotes

### PR DESCRIPTION
**Pull request by bwindels**
Fixes https://github.com/vector-im/riot-web/issues/9863 & https://github.com/vector-im/riot-web/issues/9859

edit-list-quotes

### CODE REVIEW

**Comment by aaronraimist** [comment]
Might as well fix https://github.com/vector-im/riot-web/issues/9856 while you are editing this

**Comment by bwindels** [comment]
> Might as well fix [vector-im/riot-web#9856](https://github.com/vector-im/riot-web/issues/9856) while you are editing this

Just tried and our markdown parser doesn't seem to like font tags, so will postpone this.

File: src/editor/deserialize.js [hunk 1]
```diff
@@ -17,32 +17,125 @@ limitations under the License.
...
+                return new PlainPart(`[${a.textContent}](${href})`);
             }
-            break;
+        }
+    }
+}
+
+function parseCodeBlock(n) {
+    const parts = [];
+    const preLines = `\`\`\`\n${n.textContent}\`\`\``.split("\n");
```
Comment (jryans) [inline src/editor/deserialize.js]: What is this trying to do...? Why so many backticks...? 😕

Comment (bwindels) [reply src/editor/deserialize.js]: I guess this would be nicer as "```\n" + ...

**Pull request merged**